### ISSUE TITLE
 I don't know why I called his optimizer of MTAdam directly, but my model has no trace of training， even if I change the gradient

### ISSUE THREAD

**Issue by HuangZR-stu** [ItzikMalkiel/MTAdam#3]
def train_model(model: nn.Module,
                trainloader: DataLoader,
                optimizer: optim.Optimizer,
                optimizer_lr: float,
                epochs: int,
                train_on_gpu: bool,
                ):
    r"""A function to train model,support using gpu to accelerate.

    Args:
        model (nn.Module): torch.nn.Module
        trainloader (DataLoader): torch.utils.data.Dataloader
        criterion (nn.Module): torch.nn.Module
        optimizer (optim.Optimizer): torch.optim.Optimizer
        optimizer_lr (float): learning rate
        epochs (int): training epochs
        train_on_gpu (bool): whether use gpu to accelerate or not

    Returns:
        [torch.nn.Module]: return the trained model on cpu.
    """
    seed_it(19)
    if train_on_gpu:
        CUDA = cuda.is_available()
        device = ('cuda:0' if CUDA else 'cpu')
        model = model.to(device)
        if CUDA:
            print('Using gpu to accelerate ..., ', torch.cuda.get_device_name(0))
        else:
            print('Using cpu ...')
    criterion = nn.MSELoss()
    total_loss = []
    loss_f_list = []
    loss_s_list = []
    loss_list = []
    n_tasks = 2
    optimizer = MTAdam(model.parameters(), lr=optimizer_lr)

    loss_weights = np.full(2, 1 /2)  # (K,)

    for epoch in range(epochs):
        # pbar.set_description("Epoch:%d/%d" % (epoch, epochs))
        iter_count = 0

        for inputs, y_history, target in trainloader:
            iter_count += 1

            if train_on_gpu:
                inputs = inputs.to(device)
                y_history = y_history.to(device)
                target = target.to(device)

            out_t = model(inputs, y_history)

            loss_f = criterion(out_t, target)
            loss_s = model.loss_s
            task_loss = []
            task_loss.append(loss_f)
            task_loss.append(model.loss_s)
            task_loss = torch.stack(task_loss)

            weighted_task_loss = torch.mul(torch.tensor(loss_weights).to(device), task_loss)
            loss = torch.sum(weighted_task_loss)
            ranks = [1] * n_tasks

            optimizer.step(weighted_task_loss, ranks, None)
            total_loss.append(loss.detach().item())
            loss_f_list.append(loss_f.detach().item())
            loss_s_list.append(loss_s.detach().item())
This is the code from my training phase. Is there anything wrong with what was written in the middle?
